lms/djangoapps/instructor_task/rest_api/v1/views.py

```diff
@@ -78,11 +94,103 @@ def destroy(self, request, *args, **kwargs):
         except InstructorTaskSchedule.DoesNotExist:
             return Response(status=status.HTTP_404_NOT_FOUND)
 
-        # update the task's status to REVOKED and then delete the task schedule instance
         task = schedule.task
+        # verify the task hasn't already been processed before revoking
+        try:
+            self._verify_task_state(task)
+        except TaskUpdateException as task_update_err:
+            return Response(
+                {"detail": str(task_update_err)},
+                status=status.HTTP_400_BAD_REQUEST
+            )
```

Comment:
Added this to try and safeguard against someone who has a stale page open and tries to delete a schedule for a task that is being/been executed by Celery.